python/pyarrow/table.pxi

```diff
@@ -1016,6 +1058,21 @@ cdef class RecordBatch(_PandasConvertible):
             entries.append((name, column))
         return ordered_dict(entries)
 
+    def to_pylist(self):
+        """
+        Convert the RecordBatch to a list of dictionaries.
+
+        Returns
+        -------
+        list
+        """
+        entries = []
+        for i in range(self.batch.num_columns()):
+            name = bytes(self.batch.column_name(i)).decode('utf8')
+            column = self[i].to_pylist()
+            entries.append({name: column})
```

Comment:
Given this might involve thousand of columns I think there is value in building it as a list comprehension, it's generally faster than multiple appends.